repos/system_upgrade/el7toel8/actors/localreposinhibit/actor.py

```diff
@@ -0,0 +1,64 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.models import TMPTargetRepositoriesFacts, UsedTargetRepositories
+from leapp.reporting import Report
+from leapp.tags import IPUWorkflowTag, TargetTransactionChecksPhaseTag
+
+
+class LocalReposInhibit(Actor):
+    """Inhibits the upgrade if local repositories were found."""
+
+    name = "local_repos_inhibit"
+    consumes = (
+        UsedTargetRepositories,
+        TMPTargetRepositoriesFacts,
+    )
+    produces = (Report,)
+    tags = (IPUWorkflowTag, TargetTransactionChecksPhaseTag)
+
+    def process(self):
+        # fmt: off
+        used_target_repos = next(self.consume(UsedTargetRepositories)).repos
+        target_repos = next(self.consume(TMPTargetRepositoriesFacts)).repositories
+        # fmt: on
+        target_repo_id_to_url_map = {
```

Comment:
@bocekm Michal, what do you think to leave this logic here, without moving to a separate function? It is simplified and, from my POV, looks readable like it is.

Reply:
Frankly, it's still less readable than I'd like to see.
Ideally, the condition could look like:
`if _file_baseurl_in_use()`
and put the lines 20-33 in that function.